### ISSUE TITLE
Remove react-admin dependency from data provider

### ISSUE THREAD

**Issue by srosset81** [assemblee-virtuelle/semapps#1221]
In order to use the data provider with [Astro](https://astro.build), we should remove `react-admin` and `react` from the peerDependencies of the package.

- Create a local copy of react-admin `fetchUtils.fetchJson` 
- Add `react-admin` to the devDependencies so that it can be used to import Typescript declarations ?
- Put hooks into a `@semapps/hooks` packages ?
- Remove reification components or move them to `@semapps/input-components` ?

**Comment by Laurin-W** [comment]
I don't think we should add `react-admin` to the dev dependencies for the fetch function signatures.

Instead we can use the regular fetch function's signature.
This gives us more freedom in general to process the results of a fetch and less trouble with migrations like this one.
That doesn't mean that the fetch function we provide doesn't add auth headers by default. The inrupt client libraries use this concept as well which they call `authenticatedFetch`.
For example, I wanted to fetch data in turtle format, then the `fetchJson` function internally tries to convert the turtle to json first, which I find a bit odd. This fails so `undefined` is returned instead for the `json` prop.

I agree with the package splitting idea.

**Comment by srosset81** [comment]
> I don't think we should add `react-admin` to the dev dependencies for the fetch function signatures.

This was not my proposal. My proposal was to copy the function `fetchUtils.fetchJson` locally, so that we don't need the dependency anymore. But we still have a dependency for the React-Admin types, because they are imported everywhere. So I don't think we can do without a `devDependency` to React-Admin.

The fetch function of the data provider does not only add authentication, it also redirects the query to the proxy if needed. It also adds headers if necessary.

**Comment by Laurin-W** [comment]
> The fetch function of the data provider does not only add authentication, it also redirects the query to the proxy if needed. It also adds headers if necessary.

Sure, we can add that too with an `authenticatedFetch`-like function!

> But we still have a dependency for the React-Admin types, because they are imported everywhere.

I just had a look. There are about 20 files importing `react-admin`. The ones that don't import `fetchUtils` are either importing `useDataProvider`, `RaRecord` or other react-based components.

**Comment by srosset81** [comment]
> Sure, we can add that too with an authenticatedFetch-like function!

👍 

> I just had a look. There are about 20 files importing `react-admin`. The ones that don't import `fetchUtils` are either importing `useDataProvider`, `RaRecord` or other react-based components.

Oh I was talking about the data provider, not the other hooks and components in the package. That's why I was suggesting on the first message to extract them to other packages. The data provider itself only import `fetchUtils` and React-Admin types.
